src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a === 0 || this.b === 0 || this.c === 0) {
+      throw new Error('Side of triangle cannot be 0!');
+    }
+
+    const sides: number[] = [this.a, this.b, this.c];
+
+    sides.sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('Invalid triangle!');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
 
+    const s = (a + b + c) / 2;
```

Comment:
it's not clear what `s` mean, use better logic naming , help another devs that looking on your code :)